assets/javascripts/app/update_checker.coffee

```diff
@@ -3,13 +3,13 @@ class app.UpdateChecker
     @lastCheck = Date.now()
 
     $.on window, 'focus', @onFocus
-    app.appCache.on 'updateready', @onUpdateReady if app.appCache
+    app.serviceWorker.on 'updateready', @onUpdateReady if app.serviceWorker
```

Comment:
The changed lines in this file could use the optional property access operator:

```coffee
app.serviceWorker?.on 'updateready', @onUpdateReady
app.serviceWorker?.update()
```

Reply:
Yes to the first line, no to the second line. The update() code is as follows:

```ruby
if app.serviceWorker
  app.serviceWorker.update()
else
  # check for update in another way
```

Because of the `else` clause, the optional property access operator cannot be used here.